src/kvirc/ui/KviWebPackageManagementDialog.cpp

```diff
@@ -39,19 +37,89 @@
 
 #include <QScreen>
 #include <QToolButton>
-#include <QLineEdit>
 #include <QToolTip>
-#include <QTimer>
 #include <QClipboard>
 #include <QHBoxLayout>
-#include <QWebEngineView>
 #include <QVBoxLayout>
 #include <QToolBar>
 #include <QFile>
 #include <QProgressBar>
 #include <QDir>
 #include <QUrl>
 #include <QShowEvent>
+#include <QJsonDocument>
+#include <QJsonArray>
+#include <QMessageBox>
+#include <QMenu>
+
+#define KVI_WEBPACK_INDEX "index.json"
+
+KviWebPackageListItem::KviWebPackageListItem(KviTalListWidget * pBox, QJsonObject obj, const QString & szBaseUrl)
+	: KviTalListWidgetItem(pBox)
+{
+	m_szName = obj.value("name").toString();
+	m_szVersion = obj.value("version").toString();
+	m_szAuthor = obj.value("author").toString();
+	m_szDesc = obj.value("desc").toString();
+	m_szScreen = obj.value("screen").toString();
+	m_szDownload = obj.value("download").toString();
+
+	QString szText;
+	szText = "<b>";
+	szText += m_szName;
+	szText += "</b>";
+
+	if(!m_szVersion.isEmpty())
+	{
+		szText += "[";
+		szText += m_szVersion;
+		szText += "]";
+	}
+
+	if(!m_szAuthor.isEmpty())
+	{
+		szText += " <font color=\"#454545\"> ";
+		szText += __tr2qs("by");
+		szText += " ";
+		szText += m_szAuthor;
+		szText += "</font>";
+	}
+
+	szText += "<br><font size=\"-1\">";
+	szText += m_szDesc;
+	szText += "</font>";
+	setText(szText);
+
+	if(!m_szScreen.isEmpty())
+	{
+		downloadIcon(szBaseUrl + m_szScreen);
+	}
+}
+
+void KviWebPackageListItem::downloadIcon(const QString & szIconUrl)
+{
+	QNetworkRequest req(szIconUrl);
+	QNetworkReply * pReply = KviNetworkAccessManager::getInstance()->get(req);
+	connect(pReply, &QNetworkReply::finished, [this, pReply]() {
+		if (pReply->error() == QNetworkReply::NoError) {
+			QPixmap pixImage;
+			if(pixImage.loadFromData(pReply->readAll()))
+			{
+				setIcon(pixImage);
+			}
+		}
+	});
+}
+
+void KviWebPackageListItem::showPopupImage()
+{
+	QDialog dlg;
+	QHBoxLayout *l = new QHBoxLayout(&dlg);
```

Comment:
Can you use consistent way of declaring pointers (wrt spaces)?

Probably whatever clang-format does

Reply:
Done, thank you